arangod/Cluster/Maintenance.cpp

```diff
@@ -2663,23 +2663,39 @@ void arangodb::maintenance::syncReplicatedShardsWithLeaders(
         // that we have been restarted but the leader did not notice that
         // we were gone, we must check if the leader is set correctly here
         // locally for our shard:
+        VPackSlice theLeader = VPackSlice::emptyStringSlice();
         VPackSlice lshard = localdb.get(shname);
         TRI_ASSERT(lshard.isObject());
         bool needsResyncBecauseOfRestart = false;
         if (lshard.isObject()) {  // just in case
-          VPackSlice theLeader = lshard.get(THE_LEADER);
+          theLeader = lshard.get(THE_LEADER);
           if (theLeader.isString() &&
               theLeader.stringView() ==
                   maintenance::ResignShardLeadership::LeaderNotYetKnownString) {
             needsResyncBecauseOfRestart = true;
           }
         }
 
+        if (cservers.length() == 0 ||
+            cservers[0].stringView().starts_with("_")) {
+          // Do not attempt to sync if the server in current is still resigned.
+          LOG_TOPIC("2dba7", INFO, Logger::MAINTENANCE)
+              << "refuse to synchronize shard with a resigned leader in "
```

Comment:
We haven't decided if this is how we want to fix it but for a hotfix it is good enough...

Reply:
I agree. The hot fix is to allow HPE to teston QA clusters.